### PR TITLE
[Merged by Bors] - chore: reorder lakefile to put dependencies first

### PR DESCRIPTION
**Pull request by kim-em**
The lakefile options configuration has grown long enough that the dependencies were off the first screen. Since I change these often (and, I guess logically they do come first!), I'd like to move them to the top.

### CODE REVIEW

File: lakefile.lean [hunk 1]
```diff
@@ -2,6 +2,21 @@ import Lake
 
 open Lake DSL
 
+
+/-!
+## Mathlib dependencies on upstream projects.
```
Comment (grunweg) [inline lakefile.lean:7]: Pre-existing minor nitpick: does not need to end on a full stop? For me, this is not a complete sentence
```suggestion
## Mathlib dependencies on upstream projects

**Review by grunweg — APPROVED** [review]
Sounds reasonable enough.
maintainer merge

**Comment by mattrobball** [comment]
I don't particularly like the period either. 

bors delegate+

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**